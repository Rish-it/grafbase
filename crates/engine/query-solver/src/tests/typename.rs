use crate::assert_solving_snapshots;

const SCHEMA: &str = r###"
enum join__Graph {
  PRODUCTS @join__graph(name: "products", url: "http://products:4003/graphql")
}

type Product
{
  upc: String!
}

type Query
  @join__type(graph: PRODUCTS)
{
  topProducts(first: Int = 5): [Product] @join__field(graph: PRODUCTS)
}
"###;

#[tokio::test]
async fn top_level_typename() {
    assert_solving_snapshots!(
        "top_level_typename",
        SCHEMA,
        r#"
        query {
            __typename
        }
        "#
    );
}

#[tokio::test]
async fn only_typename() {
    assert_solving_snapshots!(
        "only_typename",
        SCHEMA,
        r#"
        {
          topProducts {
            __typename
          }
        }
        "#
    );
}
