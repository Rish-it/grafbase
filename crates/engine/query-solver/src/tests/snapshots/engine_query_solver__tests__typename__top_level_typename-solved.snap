---
source: crates/engine/query-solver/src/tests/typename.rs
expression: "digraph {\n    0 [ label = \"root\", color=forestgreen ]\n    1 [ label = \"__typename\", color=forestgreen ]\n    2 [ label=\"\", style=dashed]\n    0 -> 1 [ label = \"\", color=forestgreen,fontcolor=forestgreen ]\n    2 -> 0 [ label = \"\", color=royalblue,fontcolor=royalblue,style=dashed ]\n}\n"
---
digraph {
    0 [ label = "root", steiner=1 ]
    1 [ label = "__typename", steiner=1 ]
    2 [ label="", style=dashed]
    0 -> 1 [ cost=0, steiner=1]
    2 -> 0 [ cost=0, steiner=0]
}
